src/ApiService/ApiService/Log.cs

```diff
@@ -0,0 +1,172 @@
+        if (caller is not null) tags.Add("CalledBy", caller);
+        telemetryClient.TrackTrace(message, level, tags);
+    }
+    public void LogEvent(Guid correlationId, String evt, IDictionary<string, string> tags, IDictionary<string, double>? metrics, string? caller) {
+        tags.Add("Correlation ID", correlationId.ToString());
+        if (caller is not null) tags.Add("CalledBy", caller);
+        telemetryClient.TrackEvent(evt, properties: tags, metrics: metrics);
+    }
+    public void LogException(Guid correlationId, Exception ex, IDictionary<string, string> tags, IDictionary<string, double>? metrics, string? caller) {
+        tags.Add("Correlation ID", correlationId.ToString());
+
+        if (caller is not null) tags.Add("CalledBy", caller);
+        telemetryClient.TrackException(ex, tags, metrics);
+    }
+
+    public void Flush() {
+        telemetryClient.Flush();
+    }
+}
+
+//TODO: Should we write errors and Exception to std err ? 
+class Console : ILog {
+
+    private string DictToString<T>(IDictionary<string, T>? d) {
+        if (d is null)
+        {
+            return string.Empty;
+        }
+        else
+        {
+            return string.Join("", d);
+        }
+    }
+
+    private void LogTags(Guid correlationId, string? caller, IDictionary<string, string> tags) {
+        var ts = DictToString(tags);
+        if (!string.IsNullOrEmpty(ts))
+        {
+            System.Console.WriteLine("[{0}:{1}] Tags:{2}", correlationId, caller, ts);
+        }
+    }
+
+    private void LogMetrics(Guid correlationId, string? caller, IDictionary<string, double>? metrics) {
+        var ms = DictToString(metrics);
+        if (!string.IsNullOrEmpty(ms)) {
+            System.Console.Out.WriteLine("[{0}:{1}] Metrics:{2}", correlationId, caller, DictToString(metrics));
+        }
+    }
+
+    public void Log(Guid correlationId, String message, SeverityLevel level, IDictionary<string, string> tags, string? caller) {
+        System.Console.Out.WriteLine("[{0}:{1}][{2}] {3}", correlationId, caller, level, message);
+        LogTags(correlationId, caller, tags);
+    }
+
+    public void LogEvent(Guid correlationId, String evt, IDictionary<string, string> tags, IDictionary<string, double>? metrics, string? caller) {
+        System.Console.Out.WriteLine("[{0}:{1}][Event] {2}", correlationId, caller, evt);
+        LogTags(correlationId, caller, tags);
+        LogMetrics(correlationId, caller, metrics);
+    }
+    public void LogException(Guid correlationId, Exception ex, IDictionary<string, string> tags, IDictionary<string, double>? metrics, string? caller) {
+        System.Console.Out.WriteLine("[{0}:{1}][Exception] {2}", correlationId, caller, ex);
+        LogTags(correlationId, caller, tags);
+        LogMetrics(correlationId, caller, metrics);
+    }
+    public void Flush() {
+        System.Console.Out.Flush();
+    }
+}
+
+public class Log {
+
+    static List<ILog> loggers;
+
+    static Log() {
```

Comment:
Can we have the logging be an instance method. And use dependency injection to pass around the current logger.

Reply:
probably, 

we can look into this later. so far there is no code that dependent on this, so nothing will be broken with your proposed change